demo_nodes_cpp/CMakeLists.txt

```diff
@@ -11,9 +11,12 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
I am not convinced we should recommend users to use this approach. I think there is value in using the common visibility macros instead.